src/components/AddPaymentCard/PaymentCardChangeCurrencyForm.tsx

```diff
@@ -0,0 +1,160 @@
+import React, {useCallback, useMemo, useState} from 'react';
+import {View} from 'react-native';
+import type {ValueOf} from 'type-fest';
+import FormProvider from '@components/Form/FormProvider';
+import InputWrapper from '@components/Form/InputWrapper';
+import type {FormInputErrors, FormOnyxValues} from '@components/Form/types';
+import MenuItemWithTopDescription from '@components/MenuItemWithTopDescription';
+import SelectionList from '@components/SelectionList';
+import RadioListItem from '@components/SelectionList/RadioListItem';
+import Text from '@components/Text';
+import TextInput from '@components/TextInput';
+import TextLink from '@components/TextLink';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import * as ValidationUtils from '@libs/ValidationUtils';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import INPUT_IDS from '@src/types/form/ChangeBillingCurrencyForm';
+import PaymentCardCurrencyModal from './PaymentCardCurrencyModal';
+
+type PaymentCardFormProps = {
+    initialCurrency?: ValueOf<typeof CONST.CURRENCY>;
+    isSecurityCodeRequired?: boolean;
+    changeBillingCurrency: (currency?: ValueOf<typeof CONST.CURRENCY>, values?: FormOnyxValues<typeof ONYXKEYS.FORMS.CHANGE_BILLING_CURRENCY_FORM>) => void;
+};
+
+const REQUIRED_FIELDS = [INPUT_IDS.SECURITY_CODE];
+
+function HeaderCurrencyInfo({isSectionList}: {isSectionList?: boolean}) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+    return (
+        <View style={[isSectionList && styles.mh5]}>
+            <Text style={[styles.mt3, isSectionList && styles.mb5]}>
+                {`${translate('billingCurrency.note')}`}{' '}
+                <TextLink
+                    style={styles.link}
+                    href={CONST.PRICING}
+                >{`${translate('billingCurrency.noteLink')}`}</TextLink>{' '}
+                {`${translate('billingCurrency.notDetails')}`}
+            </Text>
+        </View>
+    );
+}
```

Comment:
Any particular reason this is not in its own file? I'm unaware if this is common practice in our codebase. I tend to like to put components into their own files just to make it easier to reuse them later and also to make files generally smaller. Let me know your thoughts.